packages/form/src/TextInputRow.js

```diff
@@ -59,32 +87,63 @@ class TextInputRow extends PureComponent {
         this.props.onBlur(event);
     }
 
-    renderInput(inputProps) {
+    handleInputChange = (event) => {
+        this.updateInputHeight(event.target);
+        this.props.onChange(event);
+    }
+
+    renderInput({
+        multiLine,
+        style: inputStyle = {},
+        ...inputProps
+    }) {
+        const sharedProps = {
+            ref: this.setInputRef,
+            className: BEM.input.toString(),
+            onFocus: this.handleInputFocus,
+            onBlur: this.handleInputBlur,
+        };
+
+        const textareaStyle = {
+            ...inputStyle,
+            height: this.state.inputHeight,
+        };
+
+        if (multiLine) {
+            return (
+                <textarea
+                    style={textareaStyle}
+                    onChange={this.handleInputChange}
+                    {...sharedProps}
+                    {...inputProps} />
+            );
+        }
+
         return (
             <input
-                ref={(ref) => { this.inputRef = ref; }}
                 type="text"
-                className={BEM.input.toString()}
-                onFocus={this.handleInputFocus}
-                onBlur={this.handleInputBlur}
+                onChange={this.props.onChange}
+                {...sharedProps}
```

Comment:
Is it necessary to call the `updateInputHeight` when the normal input is focused?

Reply:
It's for this scenario: the `<TextInputRow>` is mounted with an initial paragraph. The height is adjust for the first time on mount. Then when you focus the input, its font size is increased and thus the inner `<textarea>` might be not tall enough to display all content in new font size.

At this time point, it seems necessary to grow the height again.

Reply:
...which reminds me, it should also grow if its `value` has changed.

Reply:
I see. Sorry I didn't notice the font size is adjusted when the input is focused.